pir/cpp/benchmark.cpp

```diff
@@ -0,0 +1,115 @@
+#include "benchmark/benchmark.h"
+
+#include "client.h"
+#include "server.h"
+
+namespace pir {
+namespace {
+
+std::vector<std::int64_t> generateDB(std::size_t dbsize) {
+  std::vector<std::int64_t> db(dbsize, 0);
+
+  std::generate(db.begin(), db.end(), [n = 0]() mutable {
+    ++n;
+    return 4 * n;
+  });
+
+  return db;
+}
+
+void BM_ServerLoad(benchmark::State& state) {
+  std::size_t dbsize = state.range(0);
+  auto db = generateDB(dbsize);
+  int64_t elements_processed = 0;
+
+  auto params = PIRParameters::Create(db.size());
+  auto pirdb = PIRDatabase::Create(db, params).ValueOrDie();
+
+  for (auto _ : state) {
+    auto server_ = PIRServer::Create(pirdb, params).ValueOrDie();
```

Comment:
Creating the server should be pretty trivial. Creating the database is another story.